### PR TITLE
[GEOT-6837] Migrate anonymous inner classes to lambdas, whenever it easier to read, or at least, not worse.

### PR DESCRIPTION
**Pull request by aaime**
<Include a few sentences describing the overall goals for this Pull Request>

# Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

For all pull requests:

- [ ] Confirm you have read the [contribution guidelines](https://github.com/geotools/geotools/blob/master/CONTRIBUTING.md)
- [ ] You have sent a Contribution Licence Agreement (CLA) as necessary (not required for small changes, e.g., fixing typos in documentation)
- [ ] Make sure the first PR targets the master branch, eventual backports will be managed later. This can be ignored if the PR is fixing an issue that only happens in a specific branch, but not in newer ones.
- [ ] The changes are not breaking the build in downstream projects using SNAPSHOT dependencies, GeoWebCache and GeoServer (there is an automatic PR check verifying this, check this when it turns green).

The following are required only for core and extension modules (they are welcomed, but not required, for unsupported modules):

- [ ] There is an issue in [Jira](https://osgeo-org.atlassian.net/projects/GEOT) describing the bug/task/new feature (a notable exemptions is, changes not visible to end users). The ticket is for the GeoTools project, if the issue was found elsewhere it's a good practice to link to the origin ticket/issue.
- [ ] The pull request contains changes related to a single objective. If multiple focuses cannot be avoided, each one is in its own commit and has a separate ticket describing it.
- [ ] PR for bug fixes and small new features are presented as a single commit
- [ ] Commit message(s) must be in the form "[GEOT-XYZW] Title of the Jira ticket"
- [ ] New unit tests have been added covering the changes
- [ ] This PR passes all existing unit tests (test results will be reported by Continuous Integration after opening this PR)
- [ ] This PR passes the [QA checks](https://docs.geotools.org/latest/developer/conventions/code/qa.html) (QA checks results will be reported by Continuous Integration after opening this PR)
- [ ] Documentation has been updated accordingly.

Submitting the PR does not require you to check all items, but by the time it gets merged, they should be either satisfied or not applicable.

### CODE REVIEW

File: build/maven/xmlcodegen/src/main/java/org/geotools/maven/xmlcodegen/AbstractGeneratorMojo.java [hunk 1]
```diff
@@ -299,100 +299,87 @@ protected XSDSchema schema() {
...
-                        }
-                    }
-                }
-             
-                getLog().warn( "XSDSchemaLocator: Could not locate schema for: " + namespaceURI );
-                return null;
-            }
-		    
-		};
+		XSDSchemaLocator locator = (schema, namespaceURI, rawSchemaLocationURI,
```
Comment (aaime) [inline build/maven/xmlcodegen/src/main/java/org/geotools/maven/xmlcodegen/AbstractGeneratorMojo.java:302]: I personally find these two pretty ugly too, but they were bad in the original as well.. such a large anonymous class shold probably be turned into an inner class.

Comment (jodygarnett) [reply build/maven/xmlcodegen/src/main/java/org/geotools/maven/xmlcodegen/AbstractGeneratorMojo.java:302]: Yep a net zero improvement on this one.

File: build/maven/xmlcodegen/src/main/java/org/geotools/maven/xmlcodegen/CycleSchemaGenerator.java [hunk 2]
```diff
@@ -235,12 +235,7 @@ protected String getSchemaClassTemplateName() {
     public List<AttributeType> sort() {
         List<AttributeType> sorted = new ArrayList<>(
                 types.values());
-        Collections.sort(sorted, new Comparator<AttributeType>() {
-            @Override
-            public int compare(AttributeType at1, AttributeType at2) {
-                return ((NameImpl) at1.getName()).compareTo((NameImpl) at2.getName());
-            }
-        });
+        Collections.sort(sorted, (at1, at2) -> ((NameImpl) at1.getName()).compareTo((NameImpl) at2.getName()));
```
Comment (jodygarnett) [inline build/maven/xmlcodegen/src/main/java/org/geotools/maven/xmlcodegen/CycleSchemaGenerator.java:238]: This is more what I was expecting, is the cast to NameImpl needed?

File: docs/src/main/java/org/geotools/graph/GraphExamples.java [hunk 3]
```diff
@@ -45,13 +43,7 @@ void graphExample() throws Exception {
...
         SimpleFeatureCollection fc = featureSource.getFeatures();
 
-        fc.accepts(
-                new FeatureVisitor() {
-                    public void visit(Feature feature) {
-                        generator.add(feature);
-                    }
-                },
-                null);
+        fc.accepts(feature -> generator.add(feature), null);
```
Comment (jodygarnett) [inline docs/src/main/java/org/geotools/graph/GraphExamples.java:46]: all of these feature visitor ones are great.

**Comment by jodygarnett** [comment]
I like it Andra, a +1 from me.

File: build/maven/xmlcodegen/src/main/java/org/geotools/maven/xmlcodegen/CycleSchemaGenerator.java (hunk 2 above)
Comment (aaime) [reply build/maven/xmlcodegen/src/main/java/org/geotools/maven/xmlcodegen/CycleSchemaGenerator.java:238]: No idea, the xmlcodegen modules has been written by someone else... that said, the cast to NameImpl was already in the orginal code. I'll notice NameImpl is not GeoTools one, it does not even have the "getName" method.

File: docs/src/main/java/org/geotools/graph/GraphExamples.java (hunk 3 above)
Comment (aaime) [reply docs/src/main/java/org/geotools/graph/GraphExamples.java:46]: Yep, they are the "low hanging fruit" of this PR indeed

**Comment by aaime** [comment]
Updated to latest main and cleaned up the cases where it did not provide benefits

**Pull request merged**